drivers/SmartThings/sonos/src/lifecycle_handlers.lua

```diff
@@ -0,0 +1,124 @@
+local capabilities = require "st.capabilities"
+
+local cosock = require "cosock"
+local log = require "log"
+local utils = require "utils"
+local PlayerFields = require "fields".SonosPlayerFields
+
+---@class SonosDriverLifecycleHandlers
+local SonosDriverLifecycleHandlers = {}
+
+local function emit_component_event_no_cache(device, component, capability_event)
+  if not device:supports_capability(capability_event.capability, component.id) then
+    local err_msg = string.format("Attempted to generate event for %s.%s but it does not support capability %s",
+      device.id, component.id, capability_event.capability.NAME)
+    log.warn_with({ hub_logs = true }, err_msg)
+    return false, err_msg
+  end
+  local event, err = capabilities.emit_event(device, component.id, device.capability_channel, capability_event)
+  if err ~= nil then
+    log.warn_with({ hub_logs = true }, err)
+  end
+  return event, err
+end
+
+---@param driver SonosDriver
+---@param device SonosDevice
+function SonosDriverLifecycleHandlers.initialize_device(driver, device)
+  -- Remove usage of the state cache for sonos devices to avoid large datastores
+  device:set_field("__state_cache", nil, { persist = true })
+  device:extend_device("emit_component_event", emit_component_event_no_cache)
+
+  device:emit_event(capabilities.mediaPlayback.supportedPlaybackCommands({
+    capabilities.mediaPlayback.commands.play.NAME,
+    capabilities.mediaPlayback.commands.pause.NAME,
+    capabilities.mediaPlayback.commands.stop.NAME,
+  }))
+
+  device:emit_event(capabilities.mediaTrackControl.supportedTrackControlCommands({
+    capabilities.mediaTrackControl.commands.nextTrack.NAME,
+    capabilities.mediaTrackControl.commands.previousTrack.NAME,
+  }))
+
+  if not driver:has_received_startup_state() then
+    device.log.debug("Driver startup state not yet received, delaying initialization of device.")
+    driver:queue_device_init_for_startup_state(device)
+    return
+  end
+
+  -- spawn a task to handle initialization to avoid blocking the main driver or device
+  -- threads, as this may involve long-yielding operations.
+  cosock.spawn(function()
+    device.log.debug("top of init loop")
+    local mac_addr = device.device_network_id
+    local player_info_tx, player_info_rx = cosock.channel.new()
+    while true do
+      device.log.debug("sending player info request")
+      driver.ssdp_task:get_player_info(player_info_tx, mac_addr)
+      device.log.debug("selecton on player info receive")
+      local recv_ready, _, select_err = cosock.socket.select({ player_info_rx }, nil, nil)
```

Comment:
Im guessing its been this way and there aren't any changes, just a moving of code, but why do we use select for one socket? Wouldn't calling `player_info_rx:receive()` have the same effect?